resilience4j-ratpack/src/main/java/io/github/resilience4j/ratpack/bulkhead/BulkheadMethodInterceptor.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2017 Jan Sykora
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.resilience4j.ratpack.bulkhead;
+
+import com.google.inject.Inject;
+import io.github.resilience4j.bulkhead.BulkheadFullException;
+import io.github.resilience4j.bulkhead.BulkheadRegistry;
+import io.github.resilience4j.bulkhead.operator.BulkheadOperator;
+import io.github.resilience4j.ratpack.recovery.RecoveryFunction;
+import io.reactivex.Flowable;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import ratpack.exec.Promise;
+import ratpack.util.Exceptions;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * A {@link MethodInterceptor} to handle all methods annotated with {@link Bulkhead}. It will
+ * handle methods that return a Promise only. It will add a transform to the promise with the bulkhead and
+ * fallback found in the annotation.
+ */
+public class BulkheadMethodInterceptor implements MethodInterceptor {
+
+    @Inject(optional = true)
+    private BulkheadRegistry registry;
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Object invoke(MethodInvocation invocation) throws Throwable {
+        Bulkhead annotation = invocation.getMethod().getAnnotation(Bulkhead.class);
+        RecoveryFunction<?> recoveryFunction = annotation.recovery().newInstance();
+        if (registry == null) {
```

Comment:
Is there any real case where `registry` won't be injected before first `invoke` execution.

Reply:
I kept the implementation same as other interceptors. But I don't think there are many cases, when this would occur. I think most use cases for these annotations are in with methods that make external calls(and requests are already being handled), which means that registry(context) is already created. The creation of context happens when Ratpack initializes Netty internals and adds handler adapter(contains all Ratpack handlers and registry) as last channel inbound handler. https://github.com/ratpack/ratpack/blob/master/ratpack-core/src/main/java/ratpack/server/internal/DefaultRatpackServer.java#L263

Reply:
I think it's fine if the implementation is similar to the other interceptors.